pkg/reconciler/ingress/ingress.go

```diff
@@ -265,6 +262,33 @@ func (r *Reconciler) FinalizeKind(ctx context.Context, ing *v1alpha1.Ingress) pk
 		}
 	}
 
+	return r.reconcileDeletion(ctx, ing)
+}
+
+func (r *Reconciler) reconcileDeletion(ctx context.Context, ing *v1alpha1.Ingress) error {
+	if !r.shouldReconcileTLS(ing) {
+		return nil
+	}
+
+	for _, tls := range ing.Spec.TLS {
+		nameNamespaces, err := resources.GetIngressGatewaySvcNameNamespaces(ctx)
+		if err != nil {
+			return err
+		}
+		for _, nameNamespace := range nameNamespaces {
+			secrets, err := r.GetSecretLister().Secrets(nameNamespace.Namespace).List(labels.SelectorFromSet(
+				resources.MakeTargetSecretLabels(tls.SecretName, tls.SecretNamespace)))
+			if err != nil {
+				return err
+			}
+			for _, secret := range secrets {
+				if err := r.GetKubeClient().CoreV1().Secrets(secret.Namespace).Delete(secret.Name, &metav1.DeleteOptions{}); err != nil {
```

Comment:
Do we want to continue deleting?

Reply:
I think we may want. Updated the code to try deleting the rest of secrets when error happens.